### PR TITLE
Documentation: updated the sample code in softmax_regression.hpp.

### PR DESCRIPTION
**Pull request by AYESDIE**
Seems like the sample code for `mlpack::regression::SoftmaxRegression` in `softmax_regression.hpp` were outdated.

### CODE REVIEW

File: src/mlpack/methods/softmax_regression/softmax_regression.hpp [hunk 1]
```diff
@@ -36,27 +36,24 @@ namespace regression {
  *
  * @code
  * arma::mat train_data; // Training data matrix.
```
Comment (zoq) [inline src/mlpack/methods/softmax_regression/softmax_regression.hpp]: I know you didn't wrote that part but do you mind to change the style to ``trainData``, just to match with the rest of the codebase?

File: src/mlpack/methods/softmax_regression/softmax_regression.hpp [hunk 2]
```diff
@@ -36,27 +36,24 @@ namespace regression {
...
- * // Use an instantiated optimizer for the training.
- * SoftmaxRegressionFunction srf(train_data, labels, inputSize, numClasses);
- * L_BFGS<SoftmaxRegressionFunction> optimizer(srf, numBasis, numIterations);
- * SoftmaxRegression<L_BFGS> regressor2(optimizer);
+ * // Train the model using an instantiated optimizer for the training.
+ * SoftmaxRegression regressor(train_data.n_rows, numClasses);
+ * ens::L_BFGS optimizer(numBasis, numIterations);
+ * regressor.Train(train_data, labels, numClasses, std::move(optimizer));
  *
  * arma::mat test_data; // Test data matrix.
```
Comment (zoq) [inline src/mlpack/methods/softmax_regression/softmax_regression.hpp]: See comment above.

File: src/mlpack/methods/softmax_regression/softmax_regression.hpp [hunk 3]
```diff
@@ -35,28 +35,25 @@ namespace regression {
...
  *
  * const size_t numBasis = 5; // Parameter required for L-BFGS algorithm.
  * const size_t numIterations = 100; // Maximum number of iterations.
  *
- * // Use an instantiated optimizer for the training.
- * SoftmaxRegressionFunction srf(train_data, labels, inputSize, numClasses);
- * L_BFGS<SoftmaxRegressionFunction> optimizer(srf, numBasis, numIterations);
- * SoftmaxRegression<L_BFGS> regressor2(optimizer);
+ * // Train the model using an instantiated optimizer for the training.
+ * SoftmaxRegression regressor(train_data.n_rows, numClasses);
```
Comment (zoq) [inline src/mlpack/methods/softmax_regression/softmax_regression.hpp]: Can you change this as well?

File: src/mlpack/methods/softmax_regression/softmax_regression.hpp [hunk 4]
```diff
@@ -35,28 +35,25 @@ namespace regression {
...
  *
- * arma::mat test_data; // Test data matrix.
- * arma::vec predictions1, predictions2; // Vectors to store predictions in.
+ * arma::mat testData; // Test data matrix.
+ * arma::Row<size_t> predictions; // Vectors to store predictions in.
  *
  * // Obtain predictions from both the learned models.
- * regressor1.Classify(test_data, predictions1);
- * regressor2.Classify(test_data, predictions2);
+ * regressor.Classify(test_data, predictions);
```
Comment (zoq) [inline src/mlpack/methods/softmax_regression/softmax_regression.hpp]: Can you change this as well? I think this covers all.

File: src/mlpack/methods/softmax_regression/softmax_regression.hpp (hunk 3 above)
Comment (AYESDIE) [reply src/mlpack/methods/softmax_regression/softmax_regression.hpp]: Ah, I'm sorry that I missed these, I made those changes from the phone. I'll fix those in the morning.

**Comment by rcurtin** [comment]
Thanks! :+1:

**Pull request merged**